external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/EventHubReceiverImpl.java

```diff
@@ -65,77 +57,81 @@ public EventHubReceiverImpl(EventHubSpoutConfig config, String partitionId) {
   }
 
   @Override
-  public void open(IEventHubFilter filter) throws EventHubException {
-    logger.info("creating eventhub receiver: partitionId=" + partitionId + 
-    		", filterString=" + filter.getFilterString());
+  public void open(String offset) throws EventHubException {
+    logger.info("creating eventhub receiver: partitionId=" + partitionId +
+            ", offset=" + offset);
     long start = System.currentTimeMillis();
-    receiver = new ResilientEventHubReceiver(connectionString, entityName,
-    		partitionId, consumerGroupName, defaultCredits, filter);
-    receiver.initialize();
-    
+    try {
+      ehClient = EventHubClient.createFromConnectionStringSync(connectionString);
+      receiver = ehClient.createEpochReceiverSync(
+              consumerGroupName,
+              partitionId,
+              offset,
+              false,
+              1);
+    }catch (Exception e){
+      logger.info("Exception in creating EventhubClient"+e.toString());
+    }
     long end = System.currentTimeMillis();
     logger.info("created eventhub receiver, time taken(ms): " + (end-start));
   }
 
   @Override
-  public void close() {
+  public void close(){
     if(receiver != null) {
-      receiver.close();
+      try {
+        receiver.close().whenComplete((voidargs,error)->{
+          try{
+            if(error!=null){
+              logger.error("Exception during receiver close phase"+error.toString());
+            }
+            ehClient.closeSync();
+          }catch (Exception e){
+            logger.error("Exception during ehclient close phase"+e.toString());
+          }
+        }).get();
+      }catch (InterruptedException e){
+        logger.error("Exception occured during close phase"+e.toString());
+      }catch (ExecutionException e){
+        logger.error("Exception occured during close phase"+e.toString());
+      }
       logger.info("closed eventhub receiver: partitionId=" + partitionId );
       receiver = null;
+      ehClient =  null;
     }
   }
-  
+
+
   @Override
   public boolean isOpen() {
     return (receiver != null);
   }
 
   @Override
-  public EventData receive(long timeoutInMilliseconds) {
+  public EventDataWrap receive() {
     long start = System.currentTimeMillis();
-    Message message = receiver.receive(timeoutInMilliseconds);
+    Iterable<EventData> receivedEvents=null;
+    /*Get one message at a time for backward compatibility behaviour*/
+    try {
+      receivedEvents = receiver.receiveSync(1);
```

Comment:
>receiver [](start = 23, length = 8)

this will result in NPE if in the Open() - createReceiver call threw exception (and the try catch swallowed it)